core/src/main/scala/scalafix/util/ChangeType.scala

```diff
@@ -0,0 +1,43 @@
+package scalafix.util
+
+import scala.collection.immutable.Seq
+import scala.{meta => m}
+import scalafix.rewrite._
+
+//Provide a little structure to the replacements we will be performing
+case class ReplaceType(original: m.Type,
+                       replacement: m.Type,
+                       newString: String) {
+  def toPatch(t: m.Type): Patch = Patch.replace(t, newString)
+}
+
+class ChangeType(ast: m.Tree, ctx: RewriteCtx)(implicit sApi: SemanticApi) {
+  object DType extends Desugared[m.Type]
+  object DTerm extends Desugared[m.Term]
+
+  def partialTypeMatch(
+      rt: ReplaceType): PartialFunction[m.Tree, (m.Type, ReplaceType)] = {
+    case t @ DType(desugared)
+        if StructurallyEqual(desugared, rt.original).isRight =>
```

Comment:
I was able to use StructurallyEqual for types, but haven't figured out how to do it for methods yet without losing the arguments when I create patches.  I end up matching both the method, and the method application with args, which overwrites them when I would like to not match on application.